resources/splunk_install.rb

```diff
@@ -60,25 +50,43 @@ def package_path
 
   load_current_value do |desired|
     package desired.package
-    version_file = version_pathname(install_dir)
-    current_value_does_not_exist! unless version_file.exist?
 
-    version_data = version_file.read
-    installed_version, installed_build = version_data.match(/VERSION=(\d(?:\.\d)+).+BUILD=([\w]+)/m).captures
-    version installed_version
-    build installed_build
+    user_resource = Chef::Resource::User.new(desired.user || new_resource.user, run_context)
+    user_resource.system true
+    user_resource.manage_home true
+    user_resource.run_action :create
```

Comment:
It does seem wrong to do this here, but load_current_value is the only block of code called before every action. The alternative is to create a method in the action_class that I call from every action (I had to do something similar with post_install except after the action is run). I will probably do that, as this code shouldn't reside here, I just wish there was a better way.